src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -71,6 +75,56 @@ public void RunBenchmarks(string assemblyPath)
             s_profiler(assemblyPath, Configuration.RunId, _outputDirectory, runner, collectOutputFilesCallback);
         }
 
+        public void RunScenario(ProcessStartInfo processStartInfo, Action<PerformanceTestConfig> preIterationDel,
+            Action<PerformanceTestConfig> postIterationDel, Func<PerformanceTestConfig, ScenarioBenchmark> teardownDel)
+        {
+            int iterations = _performanceTestConfig.Iterations;
+            int timeout = (int)(_performanceTestConfig.TimeoutPerIteration.TotalMilliseconds);
+
+            for (int i = 0; i < iterations; i++)
+            {
+                preIterationDel(_performanceTestConfig);
+                using(var p = new Process())
+                {
+                    p.StartInfo = processStartInfo;
+                    p.Start();
+                    if (p.WaitForExit(timeout) == false) 
+                    {
+                        if (p != null)
+                        {
+                            p.Kill();
+                        }
+                        Console.WriteLine("Timeouted!");
+                        return;
+                    }
+                }
+                postIterationDel(_performanceTestConfig);
+            }
+
+            ScenarioBenchmark scenarioBenchmark = teardownDel(_performanceTestConfig);
+            if (scenarioBenchmark == null) 
+            {
+                throw new NullReferenceException("The Teardown Delegate should return a valid instance of ScenarioBenchmark.")
+            }
+
+            string scenarioNamespace = scenarioBenchmark.Namespace;
+
+            scenarioBenchmark.Serialize(Configuration.RunId + "-" + scenarioNamespace + ".xml");
+
+            var mdFileName = Configuration.RunId + "-" + scenarioNamespace + "-Statistics.md";
```

Comment:
The RunId is used for microbenchmarks as the full file name.  We should probably be consistent here and not add the scenario namespace to the file name.